### PR TITLE
Fix parsing of backslashes at the end of field content

### PR DESCRIPTION
**Pull request by koppor**
This refs https://github.com/JabRef/jabref/issues/9668, but does not fully fix it, but eliminates the main issue.

Additional changes:

- New round trip tests at BibEntryWriterTest  - they currently work
- Within a class, spell `BibTeX` consistently.
- `ImportHandler` should use the provided `FileUpdateMonitor`
- While working on fixing https://github.com/JabRef/jabref/issues/9668, I encountered that the BibTeX entry cannot be pasted using the clipboard. Alsof ixed
  image

Tested in the GUI:

image

image

This somehow refs https://github.com/koppor/jabref/issues/52.

---

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by koppor** [comment]
This addresses the concrete user description at https://github.com/JabRef/jabref/issues/9668.

However, there is much work to do address the concrete user request. -- In case an entry cannot be parsed, it is still dropped. See also https://github.com/JabRef/jabref/issues/9668#issuecomment-1473753353

File: src/main/java/org/jabref/logic/importer/fileformat/BibtexParser.java [hunk 1]
```diff
@@ -898,7 +911,32 @@ private StringBuilder parseBracketedTextExactly() throws IOException {
...
 
-            boolean isClosingBracket = (character == '}') && (lastCharacter != '\\');
+            boolean isClosingBracket = false;
+            if (character == '}') {
+                if (lastCharacter == '\\') {
+                    // We hit `\}`
+                    // It could be that a user has a backslash at the end of the entry, but intended to put a file path
+                    // We want to be relaxed at that case
+                    // First described at https://github.com/JabRef/jabref/issues/9668
+                    char[] peek2 = peek2();
```
Comment (calixtus) [inline src/main/java/org/jabref/logic/importer/fileformat/BibtexParser.java:921]: What does peek2 mean? What does that method name mean?

Comment (koppor) [reply src/main/java/org/jabref/logic/importer/fileformat/BibtexParser.java:921]: The existing method `peek` has a look on one next character (without consuming it). `peek2` does similar thing, but for two characters. - Both methods do not have JavaDoc comment, but readable code?

image

Maybe, I just need to rename `peek2` to `peekTwoCharacters`?

Comment (calixtus) [reply src/main/java/org/jabref/logic/importer/fileformat/BibtexParser.java:921]: Rename would be sufficient imho.

Comment (koppor) [reply src/main/java/org/jabref/logic/importer/fileformat/BibtexParser.java]: I also added some more descriptions?

**Review by Siedlerchr — APPROVED** [review]

**Review by Siedlerchr — DISMISSED** [review]

**Pull request merged**